fusion/imagedet/exp/kitti_sharing_dl_10/config.txt
torch version: 1.8.1+cu111
cudnn version: 8005
cmd: ['/home/stavros/Workspace/SqueezeDet-PyTorch/src/main.py', '--mode', 'sharing', '--exp_id', 'kitti_sharing', '--num_epochs', '10', '--dataset', 'kitti', '--load_model', '/home/stavros/Workspace/SqueezeDet-PyTorch/models/squeezedet_kitti_epoch280.pth']

config:
anchors: [[   8.    8.   34.   30.]
 [   8.    8.   75.   45.]
 [   8.    8.   38.   90.]
 ...
 [1240.  376.  194.  178.]
 [1240.  376.  283.  156.]
 [1240.  376.  381.  185.]]
anchors_per_grid: 9
arch: squeezedet
batch_size: 6
bbox_loss_weight: 6.0
chunk_sizes: [6]
class_loss_weight: 1.0
class_names: ('Car', 'Pedestrian', 'Cyclist')
data_dir: /home/stavros/Desktop/OpenPCDet/data
dataset: kitti
debug: 0
debug_dir: /home/stavros/Workspace/SqueezeDet-PyTorch/exp/kitti_sharing/debug
device: cuda
drift_prob: 0.2
dropout_prob: 0.5
exp_dir: /home/stavros/Workspace/SqueezeDet-PyTorch/exp
exp_id: kitti_sharing
flip_prob: 0.5
forbid_resize: False
gpus: [0]
gpus_str: 0
grad_norm: 5.0
input_size: (384, 1248)
keep_top_k: 64
load_model: /home/stavros/Workspace/SqueezeDet-PyTorch/models/squeezedet_kitti_epoch280.pth
lr: 0.01
master_batch_size: 6
mode: sharing
momentum: 0.9
negative_score_loss_weight: 100.0
nms_thresh: 0.4
no_eval: False
not_cuda_benchmark: False
num_anchors: 16848
num_classes: 3
num_epochs: 10
num_iters: -1
num_workers: 4
positive_score_loss_weight: 3.75
print_interval: 10
rgb_mean: [[[93.877 98.801 95.923]]]
rgb_std: [[[78.782 80.13  81.2  ]]]
root_dir: /home/stavros/Workspace/SqueezeDet-PyTorch
save_dir: /home/stavros/Workspace/SqueezeDet-PyTorch/exp/kitti_sharing
save_intervals: 1
score_thresh: 0.3
seed: 42
val_intervals: 5
weight_decay: 0.0001


fusion/imagedet/exp/kitti_sharing_dl_10/logs_2022-06-11-01-46/config.txt
torch version: 1.8.1+cu111
cudnn version: 8005
cmd: ['/home/stavros/Workspace/SqueezeDet-PyTorch/src/main.py', '--mode', 'sharing', '--exp_id', 'kitti_sharing', '--num_epochs', '10', '--dataset', 'kitti', '--load_model', '/home/stavros/Workspace/SqueezeDet-PyTorch/models/squeezedet_kitti_epoch280.pth']

config:
anchors: [[   8.    8.   34.   30.]
 [   8.    8.   75.   45.]
 [   8.    8.   38.   90.]
 ...
 [1240.  376.  194.  178.]
 [1240.  376.  283.  156.]
 [1240.  376.  381.  185.]]
anchors_per_grid: 9
arch: squeezedet
batch_size: 6
bbox_loss_weight: 6.0
chunk_sizes: [6]
class_loss_weight: 1.0
class_names: ('Car', 'Pedestrian', 'Cyclist')
data_dir: /home/stavros/Desktop/OpenPCDet/data
dataset: kitti
debug: 2
debug_dir: /home/stavros/Workspace/SqueezeDet-PyTorch/exp/kitti_sharing/debug
device: cuda
drift_prob: 0.2
dropout_prob: 0.5
exp_dir: /home/stavros/Workspace/SqueezeDet-PyTorch/exp
exp_id: kitti_sharing
flip_prob: 0.5
forbid_resize: False
gpus: [0]
gpus_str: 0
grad_norm: 5.0
input_size: (384, 1248)
keep_top_k: 64
load_model: /home/stavros/Workspace/SqueezeDet-PyTorch/models/squeezedet_kitti_epoch280.pth
lr: 0.01
master_batch_size: 6
mode: sharing
momentum: 0.9
negative_score_loss_weight: 100.0
nms_thresh: 0.4
no_eval: False
not_cuda_benchmark: False
num_anchors: 16848
num_classes: 3
num_epochs: 10
num_iters: -1
num_workers: 4
positive_score_loss_weight: 3.75
print_interval: 10
rgb_mean: [[[93.877 98.801 95.923]]]
rgb_std: [[[78.782 80.13  81.2  ]]]
root_dir: /home/stavros/Workspace/SqueezeDet-PyTorch
save_dir: /home/stavros/Workspace/SqueezeDet-PyTorch/exp/kitti_sharing
save_intervals: 1
score_thresh: 0.3
seed: 42
val_intervals: 5
weight_decay: 0.0001


imagedet/SqueezeDet-PyTorch/src/utils/config.py
import argparse
import os
import json

def readConfigurationFile(filepath):
    config = json.load(open(filepath, ))
    config = config['squeezedet']
    return config


class Config(object):
    def __init__(self):
        self.parser = argparse.ArgumentParser()

        # basic experiment setting
        self.parser.add_argument('--mode',
                                 help='train | eval | demo')
        self.parser.add_argument('--dataset', default='kitti',
                                 help='coco | kitti')
        self.parser.add_argument('--load_model', default='',
                                 help='path to pre-trained model')
        self.parser.add_argument('--debug', type=int, default=0,
                                 help='0: show nothing\n'
                                      '1: visualize pre-processed image and boxes\n'
                                      '2: visualize detections.')
        self.parser.add_argument('--exp_id', default='default')

        self.parser.add_argument('--external_config', default=None)

        # model
        self.parser.add_argument('--arch', default='squeezedet',
                                 help='model architecture: squeezedet | squeezedetplus')
        self.parser.add_argument('--dropout_prob', type=float, default=0.5,
                                 help='probability of dropout.')

        # train
        self.parser.add_argument('--lr', type=float, default=0.01,
                                 help='learning rate for batch size 32.')
        self.parser.add_argument('--momentum', type=float, default=0.9,
                                 help='momentum of SGD.')
        self.parser.add_argument('--weight_decay', type=float, default=0.0001,
                                 help='weight decay of SGD.')
        self.parser.add_argument('--grad_norm', type=float, default=5.,
                                 help='max norm of the gradients.')
        self.parser.add_argument('--num_epochs', type=int, default=300,
                                 help='total training epochs.')
        self.parser.add_argument('--num_iters', type=int, default=-1,
                                 help='default: #samples / batch_size.')
        self.parser.add_argument('--batch_size', type=int, default=32,
                                 help='batch size')
        self.parser.add_argument('--master_batch_size', type=int, default=-1,
                                 help='batch size on the master gpu.')
        self.parser.add_argument('--save_intervals', type=int, default=1,
                                 help='number of epochs to save model.')
        self.parser.add_argument('--val_intervals', type=int, default=5,
                                 help='number of epochs to run validation.')
        self.parser.add_argument('--no_eval', action='store_true',
                                 help='bypass mAP evaluation during training.')
        self.parser.add_argument('--print_interval', type=int, default=10,
                                 help='disable progress bar and print to screen.')
        self.parser.add_argument('--flip_prob', type=float, default=0.5,
                                 help='probability of horizontal flip during training.')
        self.parser.add_argument('--drift_prob', type=float, default=0.2,
                                 help='probability of drifting image during training.')
        self.parser.add_argument('--forbid_resize', action='store_true',
                                 help='disable image resizing during training, '
                                      'use crop/pad instead.')
        self.parser.add_argument('--class_loss_weight', type=float, default=1.,
                                 help='weight of classification loss.')
        self.parser.add_argument('--positive_score_loss_weight', type=float, default=3.75,
                                 help='positive weight of score prediction loss.')
        self.parser.add_argument('--negative_score_loss_weight', type=float, default=100.,
                                 help='negative weight of score prediction loss.')
        self.parser.add_argument('--bbox_loss_weight', type=float, default=6.,
                                 help='weight of boxes regression loss.')

        # inference
        self.parser.add_argument('--nms_thresh', type=float, default=0.4,
                                 help='discards all overlapping boxes with IoU < nms_thresh.')
        self.parser.add_argument('--score_thresh', type=float, default=0.3,
                                 help='discards all boxes with scores smaller than score_thresh.')
        self.parser.add_argument('--keep_top_k', type=int, default=64,
                                 help='keep top k detections before nms.')

        # system
        self.parser.add_argument('--gpus', default='0',
                                 help='-1 for CPU, use comma for multiple gpus')
        self.parser.add_argument('--num_workers', type=int, default=4,
                                 help='dataloader threads. 0 for single-thread.')
        self.parser.add_argument('--not_cuda_benchmark', action='store_true',
                                 help='disable when the input size is not fixed.')
        self.parser.add_argument('--seed', type=int, default=42,
                                 help='random seed')
        self.parser.add_argument('--sharing_method', default=None)
        self.parser.add_argument('--sharing_acceleration_factor',type=int, default=10)

    def parse(self, args=''):
        if args == '':
            cfg = self.parser.parse_args()
        else:
            cfg = self.parser.parse_args(args)

        # find config.json : diffenent root is defined given directory of execution
        for root, dir, files in os.walk('.'):
            if 'config.json' in files:
                configuration_file_path=os.path.join(root, 'config.json')


        if cfg.external_config is None:
            configuration_file_input=readConfigurationFile(configuration_file_path)

        if cfg.mode=='train':
            if 'load_model_training' in configuration_file_input:
                if configuration_file_input['load_model_training'] is not None:
                    if os.path.exists(configuration_file_input['load_model_training']):
                        cfg.load_model=configuration_file_input['load_model_training']

        # cfg.data_dir = os.path.join(cfg.root_dir, 'data')
        cfg.data_dir=configuration_file_input['root']+configuration_file_input['data_root']+configuration_file_input['path_to_dataset']
        cfg.gpus_str = cfg.gpus
        cfg.gpus = [int(gpu) for gpu in cfg.gpus.split(',')]
        cfg.gpus = [i for i in range(len(cfg.gpus))] if cfg.gpus[0] >= 0 else [-1]

        if cfg.mode != 'train' and len(cfg.gpus) > 1:
            print('Only single GPU is supported in {} mode.'.format(cfg.mode))
            cfg.gpus = [cfg.gpus[0]]
            cfg.master_batch_size = -1

        if cfg.master_batch_size == -1:
            cfg.master_batch_size = cfg.batch_size // len(cfg.gpus)
        rest_batch_size = (cfg.batch_size - cfg.master_batch_size)
        cfg.chunk_sizes = [cfg.master_batch_size]
        for i in range(len(cfg.gpus) - 1):
            slave_chunk_size = rest_batch_size // (len(cfg.gpus) - 1)
            if i < rest_batch_size % (len(cfg.gpus) - 1):
                slave_chunk_size += 1
            cfg.chunk_sizes.append(slave_chunk_size)
        print('training chunk_sizes:', cfg.chunk_sizes)

        cfg.root_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..'))



        cfg.exp_dir = os.path.join(cfg.root_dir, 'exp')
        cfg.save_dir = os.path.join(cfg.exp_dir, cfg.exp_id)
        cfg.debug_dir = os.path.join(cfg.save_dir, 'debug')
        print('The results will be saved to ', cfg.save_dir)
        return cfg

    @staticmethod
    def update_dataset_info(cfg, dataset):
        cfg.input_size = dataset.input_size
        cfg.rgb_mean = dataset.rgb_mean
        cfg.rgb_std = dataset.rgb_std
        cfg.class_names = dataset.class_names
        cfg.num_classes = dataset.num_classes
        cfg.anchors = dataset.anchors
        cfg.anchors_per_grid = dataset.anchors_per_grid
        cfg.num_anchors = dataset.num_anchors
        return cfg

    @staticmethod
    def print(cfg):
        names = list(dir(cfg))
        for name in sorted(names):
            if not name.startswith('_'):
                print('{:<30} {}'.format(name, getattr(cfg, name)))
